docs/tutorials/subscription-platform.md

```diff
@@ -289,6 +289,149 @@ After the initial payment during subscription creation, the recurring payments f
 
 The script will make up to a configurable number of [attempts to pay](https://developer.paypal.com/docs/archive/express-checkout/ec-set-up-reference-transactions/#capture-future-payments) an invoice before cancelling the subscription.  This attempts count is saved to the invoice itself as metadata.  The invoice's metadata is also used to prevent sending multiple failed payment emails per invoice from the PayPal payment attempts.
 
+## Google IAP Integration
+
+The Subscription Platform supports RP integrations with Google IAP (In-App Purchases).
+
+Unlike subscriptions created through the payments server website (referred to internally as "web subscriptions"), Google IAP subscriptions are managed and processed entirely outside of Stripe as required by Google. Consequently, while SubPlat maintains an awareness of subscription state in Firestore, we rely on RPs to register a Google IAP subscription for a particular FxA user when it's created, and we rely on [Real-Time Developer Notifications](https://developer.android.com/google/play/billing/rtdn-reference) to inform us of state changes, which we broadcast to RPs via the `fxa-event-broker`.
+
+### Configuration
+
+The Google IAP integration is behind a feature flag; set `subscriptions.playApiServiceAccount.enabled` in `./config/secrets.json` (see below) or the environment variable `SUBSCRIPTIONS_PLAY_API_ENABLED` to `true` before starting the auth server.
+
+In `fxa-auth-server/config/secrets.json`, set the following config values under `subscriptions`:
+
+```js
+{
+    // ...
+    "subscriptions": {
+        // ...
+        "playApiServiceAccount": {
+            "enabled": true,
+            "keyFilename": "",
+            "projectId": ""
+        }
+    }
+}
+```
+Use the same values for `keyFilename` and `projectId` as used for `authFirestore` config in [Testing IAP subscriptions locally](#testing-iap-subscriptions-locally).
+
+:::note
+To make API calls to the [Google Play Developer API](https://developer.android.com/google/play/developer-api), the service account referenced by the keyfile [must have the API enabled](https://developers.google.com/android-publisher/authorization). This is not required for most Google IAP development, so consider skipping that step.
+:::
+
+## Apple IAP Integration
+
+The Subscription Platform supports RP integrations with Apple IAP (In-App Purchases).
+
+Unlike subscriptions created through the payments server website (referred to internally as "web subscriptions"), Apple IAP subscriptions are managed and processed entirely outside of Stripe as required by Apple. Consequently, while SubPlat maintains an awareness of subscription state in Firestore, we rely on RPs to register an Apple IAP subscription for a particular FxA user when it's created, and we rely on [App Store Server notifications](https://developer.apple.com/documentation/appstoreservernotifications) to inform us of state changes, which we broadcast to RPs via the `fxa-event-broker`.
```

Comment:
This paragraph is almost identical to the intro paragraph in the "Google IAP Integration" section. It seems relevant to both, but the only other place I could think of to put it (which would be DRYer) is in the "Testing IAP subscriptions locally" section. Unfortunately, that section comes after, and content-wise makes sense to come after.

I welcome any ideas to organize this information better.

Reply:
I think an earlier version of this `md` included a brief intro to Google and Apple IAP integration. What was your motivation for removing it? I liked that approach because both intros could be combined to avoid repetition. For example:

> ## IAP Integration
> 
> The Subscription Platform supports RP integrations with both Google and Apple IAP (In-App Purchases).
> 
> Unlike subscriptions created through the payments server website (referred to internally as "web subscriptions"), both Google and Apple IAP subscriptions are managed and processed entirely outside of Stripe as required by both parties. Consequently, while SubPlat maintains an awareness of subscription state in Firestore, we rely on RPs to register Google and Apple IAP subscriptions for a particular FxA user when either are created, and we rely on [Google's Real-Time Developer Notifications](https://developer.android.com/google/play/billing/rtdn-reference) and [App Store Server notifications](https://developer.apple.com/documentation/appstoreservernotifications) to inform us of state changes, which we broadcast to RPs via the `fxa-event-broker`.

Reply:
Great question. I'm open to change this, but here is my thinking...

I opted to have one section for Google IAP and one section for Apple IAP instead of one combined section, since we have separate sections for PayPal ("Paypal Integration") for example in these docs, and the integrations are separate implementations with different configurations and setup requirements. I thought this way, we can add Google- or Apple-specific information in the future more easily.

Also the navigation menu on the right hand side in the docs only goes one level deep under each major section on the page, so I thought it'd be best if people could jump to specific sections within Apple IAP or Google IAP without having to manually scroll through the content. I.e. that nav could look either like this:
```md
# Google IAP Integration
## Configuration
## Notifications (a possible future section)
# Apple IAP Integration
## Configuration
## Notifications (a possible future section)
```
Which has all the subsections listed, compared to below, which wouldn't list any sections underneath Google IAP and Apple IAP, should more be added.
```md
# IAP Integrations
## Google IAP
## Apple IAP
```

That said they do share some similarities as well in terms of how to create test subscriptions, which has its own section; it just comes after.

Reply:
I agree with your thinking in keeping Google and Apple at top level. Their subsections should be in the navigation for easy access and if you can only go one level deep, I would rather separate them out like you did.

We could still have the section for IAP Integration (same level as Google and Apple IAP) and link to it in each section (similar to how you link to both sections in _Creating a test IAP subscription_), but in this particular case, and because we might expand on both sections in future (e.g. if there is any change on their ends), I don't think there is any harm in repeating them. Also, from a user perspective, an indivdual might only care about integrating one of the two, and too much bouncing around with anchor links can affect the overall UX.